desc/objectives/_geometry.py

```diff
@@ -459,6 +459,81 @@ def compute(self, params, constants=None):
         return data["V"]
 
 
+class CoilLength(_Objective):
+    """Coil length."""
+
+    def __init__(
+        self,
+        coil,
+        target=2 * np.pi,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        name="coil-length",
+    ):
+        super().__init__(
+            things=coil,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build objective function."""
+        coil = self.things[0]
+        self._dim_f = 1
+        # TODO: is this the correct data key?
+        # seems like it because it gives the circumference of a circular coil?
+        self._data_keys = ["length"]
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        grid = LinearGrid(L=4, M=4)
+        # TODO: what grid is supposed to be used here?
+        # TODO: am I supposed to use obj=coil?
```

Comment:
Yes, that is correct, `obj` is the type of object (or the object itself) that you are getting the profiles (or transforms, etc) for, in order to have the necessary information to calculate the data keys